tests/test_raw_node.rs

```diff
@@ -203,6 +203,61 @@ fn test_raw_node_propose_and_conf_change() {
     assert_eq!(entries[1].get_data(), &*ccdata);
 }
 
+// test_raw_node_propose_add_duplicate_node ensures that two proposes to add the same node should
+// not affect the later propose to add new node.
+#[test]
+fn test_raw_node_propose_add_duplicate_node() {
+    let s = new_storage();
+    let mut raw_node = new_raw_node(1, vec![], 10, 1, s.clone(), vec![new_peer(1)]);
+    let rd = raw_node.ready();
+    s.wl().append(&rd.entries).expect("");
+    raw_node.advance(rd);
+
+    raw_node.campaign().expect("");
+    loop {
+        let rd = raw_node.ready();
+        s.wl().append(&rd.entries).expect("");
+        if rd.ss.is_some() && rd.ss.as_ref().unwrap().leader_id == raw_node.raft.id {
+            raw_node.advance(rd);
+            break;
```

Comment:
There is only one node, which would become leader after campaign. And then it will break the loop.
It uses the same pattern to write test case like

https://github.com/pingcap/tikv/blob/master/tests/test_raw_node.rs#L176